src/transformers/trainer.py

```diff
@@ -251,6 +257,12 @@ def _is_peft_model(model):
     return False
 
 
+def _unwrap_optimizer(optimizer):
+    if isinstance(optimizer, AcceleratedOptimizer):
+        optimizer = optimizer.optimizer
+    return optimizer
```

Comment:
Is it guaranteed to only ever be one level of wrapping?

Reply:
Yes indeed ! https://github.com/huggingface/accelerate/blob/4ad4d28c49a9818e985ea12d66a89fe73fe73c87/src/accelerate/optimizer.py#L56